packages/android_alarm_manager/test/android_alarm_manager_test.dart

```diff
@@ -0,0 +1,201 @@
+      expect(
+          () => AndroidAlarmManager.oneShotAt(
+              validTime, validId, invalidCallback),
+          throwsAssertionError);
+
+      // ID should be less than 32 bits.
+      expect(
+          () => AndroidAlarmManager.oneShotAt(
+              validTime, 2147483648, validCallback),
+          throwsAssertionError);
+    });
+
+    test('sends arguments to the platform', () async {
+      final DateTime alarm = DateTime(1993);
+      const int rawHandle = 4;
+      AndroidAlarmManager.setTestOverides(
+          getCallbackHandle: (Function _) =>
+              CallbackHandle.fromRawHandle(rawHandle));
+
+      final int id = 1;
+      final bool alarmClock = true;
+      final bool allowWhileIdle = true;
+      final bool exact = true;
+      final bool wakeup = true;
+      final bool rescheduleOnReboot = true;
+
+      testChannel.setMockMethodCallHandler((MethodCall call) async {
+        expect(call.method, 'Alarm.oneShotAt');
+        expect(call.arguments[0], id);
+        expect(call.arguments[1], alarmClock);
+        expect(call.arguments[2], allowWhileIdle);
+        expect(call.arguments[3], exact);
+        expect(call.arguments[4], wakeup);
+        expect(call.arguments[5], alarm.millisecondsSinceEpoch);
+        expect(call.arguments[6], rescheduleOnReboot);
+        expect(call.arguments[7], rawHandle);
+        return true;
+      });
+
+      final bool result = await AndroidAlarmManager.oneShotAt(
+          alarm, id, (int id) => id,
+          alarmClock: alarmClock,
+          allowWhileIdle: allowWhileIdle,
+          exact: exact,
+          wakeup: wakeup,
+          rescheduleOnReboot: rescheduleOnReboot);
+
+      expect(result, isTrue);
+    });
+  });
+
+  test('${AndroidAlarmManager.oneShot} calls through to oneShotAt', () async {
+    final DateTime now = DateTime(1993);
+    const int rawHandle = 4;
+    AndroidAlarmManager.setTestOverides(
+        now: () => now,
+        getCallbackHandle: (Function _) =>
+            CallbackHandle.fromRawHandle(rawHandle));
+
+    const Duration alarm = Duration(seconds: 1);
+    final int id = 1;
+    final bool alarmClock = true;
+    final bool allowWhileIdle = true;
+    final bool exact = true;
+    final bool wakeup = true;
+    final bool rescheduleOnReboot = true;
+
+    testChannel.setMockMethodCallHandler((MethodCall call) async {
+      expect(call.method, 'Alarm.oneShotAt');
+      expect(call.arguments[0], id);
+      expect(call.arguments[1], alarmClock);
+      expect(call.arguments[2], allowWhileIdle);
+      expect(call.arguments[3], exact);
+      expect(call.arguments[4], wakeup);
+      expect(
+          call.arguments[5], now.millisecondsSinceEpoch + alarm.inMilliseconds);
+      expect(call.arguments[6], rescheduleOnReboot);
+      expect(call.arguments[7], rawHandle);
+      return true;
+    });
+
+    final bool result = await AndroidAlarmManager.oneShot(
+        alarm, id, (int id) => id,
```

Comment:
This isn't a valid callback as it's not a static named function.

Reply:
Whoops, didn't catch all of the lambdas. Thanks.